README.md

```diff
@@ -40,7 +38,318 @@ Under construction.
+
+Memphis messages are payload agnostic. Payloads are `byte[]`.
+
+In order to stop getting messages, you have to call `consumer.destroy()`. Destroy will terminate regardless
+of whether there are messages in flight for the client.
+
+### Creating a Producer
+
+```java
+Memphis memphis = new Memphis();
+Prodcuer producer = memphis.createProducer("<station-name>", "<producer-name>", false).get();
+```
+
+### Producing a message
+Without creating a producer.
+In cases where extra performance is needed the recommended way is to create a producer first
+and produce messages by using the produce function of it
+```java
+memphis.produce(
+    "<station-name>", // station_name
+    "<producer-name>", // producer_name
+    message='bytearray/protobuf class/dict/string/graphql.language.ast.DocumentNode', // bytearray / protobuf class (schema validated station - protobuf) or bytearray/dict (schema validated station - json schema) or string/bytearray/graphql.language.ast.DocumentNode (schema validated station - graphql schema)
+    false, // generate_random_suffix - defaults to false
+    15, // ack_wait_sec
+    headers, // headers - default to {}
+    false, // async_produce - defaults to false
+    "123" // msg_id
+).get();
+```
+
+### Add headers
+
+```java
+headers= new Headers()
+headers.add("key", "value")
+producer.produce(
+  message='bytearray/protobuf class/dict/string/graphql.language.ast.DocumentNode', // bytearray / protobuf class (schema validated station - protobuf) or bytearray/dict (schema validated station - json schema) or string/bytearray/graphql.language.ast.DocumentNode (schema validated station - graphql schema)
+  headers=headers).get() // default to {}
+```
+
+### Async produce
+Meaning your application won't wait for broker acknowledgement - use only in case you are tolerant for data loss
+
+```java
+producer.produce(
+  message='bytearray/protobuf class/dict/string/graphql.language.ast.DocumentNode', // bytearray / protobuf class (schema validated station - protobuf) or bytearray/dict (schema validated station - json schema) or string/bytearray/graphql.language.ast.DocumentNode (schema validated station - graphql schema)
+  headers={}, async_produce=True).get()
+```
+
+### Message ID
+Stations are idempotent by default for 2 minutes (can be configured), Idempotency achieved by adding a message id
+
+```java
+producer.produce((
+  message='bytearray/protobuf class/dict/string/graphql.language.ast.DocumentNode', // bytes / protobuf class (schema validated station - protobuf) or bytes/dict (schema validated station - json schema)
+  headers={},
+  async_produce=True,
+  msg_id="123").get()
+```
+
+### Destroying a Producer
+
+```java
+producer.destroy()
+```
+
+### Creating a Consumer
+
+```java
+Memphis memphis = new Memphis();
+Consumer consumer = memphis.createConsumer(
```

Comment:
.get() is needed I guess

Reply:
this is the python example you can't mention the param names I think, correct me if I wrong